shell/models/steve-schema.ts

```diff
@@ -0,0 +1,230 @@
+  // - resourceFields now come from a new `schemaDefinitions` endpoint
+  // - for neatness / safety / compatibility with norman resources... we fetch schemaDefinitions and return their resourceFields in a resourceFields getter
+
+  /******************
+   * Resource Fields
+   ****************** /
+
+  /**
+   * Is the property `resourceFields` available
+   *
+   * If the schema definition is required and it hasn't been fetched this will be false
+   *
+   * This is a non-erroring request, unlike the resourceFields getter which will error if schema definition is required but missing
+   */
+  get hasResourceFields(): boolean {
+    if (this.requiresResourceFields) {
+      return !!this.schemaDefinition?.resourceFields;
+    }
+
+    return !!this._resourceFields;
+  }
+
+  /**
+   * Fields associated with instances of this schema
+   *
+   * This will either come directly from the schema or from the schema's definition
+   */
+  get resourceFields(): ResourceFields {
+    if (this.requiresResourceFields) {
+      if (!this._schemaDefinitionsIds) {
+        throw new Error(`Cannot find resourceFields for Schema ${ this.id } (schemaDefinitions have not been fetched) `);
+      }
+
+      if (!this.schemaDefinition) {
+        throw new Error(`No schemaDefinition for ${ this.id } found (not in schemaDefinition response) `);
+      }
+
+      return this.schemaDefinition.resourceFields;
+    }
+
+    return this._resourceFields as ResourceFields;
+  }
+
+  /**
+   * Apply the original `resourceFields` param (if it exists). If it does not then we'll need to fetch the schema definition
+   */
+  set resourceFields(resourceFields: ResourceFields) {
+    this._resourceFields = resourceFields;
+    this.requiresResourceFields = this._resourceFields === null;
+  }
+
+  /**
+   * Ensure this schema has a populated `resourceFields` property
+   *
+   * This happens via making a request to fetch the schema definition
+   */
+  async fetchResourceFields(depth = 0): Promise<SchemaDefinition | null | undefined> {
+    if (!this.requiresResourceFields) {
+      // Not needed, no-op
+      return;
+    }
+
+    if (this.schemaDefinition) {
+      // Already have it, no-op
+      return this.schemaDefinition;
+    }
+
+    const url = this.schemaDefinitionUrl;
+
+    if (!url) {
+      console.warn(`Unable to fetch schema definitions for ${ this.id } (failed to find url of schema definition)`); // eslint-disable-line no-console
+
+      return;
+    }
+
+    if (depth >= 4) {
+      console.warn(`Unable to fetch schema definitions for ${ this.id } (too many failed requests)`); // eslint-disable-line no-console
+
+      return;
+    }
+
+    let res;
+
+    try {
+      // Make a direct request to fetch the schema definition
+      res = await this.$dispatch('request', {
+        type: STEVE.SCHEMA_DEFINITION,
+        url
+      });
+    } catch (e: any) {
+      if ( e?._status >= 500) {
```

Comment:
Why greater than 500? Probably out of your hands but....

Here's a few that seem reasonable to retry on
404,
408 Request Timeout
428 Precondition Required

And some 500s that don't:
506 Variant Also Negotiates
507 Insufficient Storage
508 Loop Detected
510 Not Extended
511 Network Authentication Required